auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java

```diff
@@ -260,4 +267,34 @@ public String getRefreshToken() {
         return refreshToken;
     }
 
+
+    /**
+     * Returns the custom attribute.
+     *
+     * @param attribute the attribute name to retrieve
+     * @return the custom attribute.
+     */
+    @Override
+    public String getCustomAttribute(final String attribute) {
+        String attributeValue = null;
+        try {
+            String identityToken = getIdentityToken();
+            JsonWebSignature signature = new JsonWebSignature();
+            signature.setCompactSerialization(identityToken);
+            String decoded = signature.getUnverifiedPayload();
+            try {
+                JSONObject rawIdentityToken = new JSONObject(decoded);
```

Comment:
it looks like keycloak supports a few types for the value: int, long, string and boolean. I think it might be better to add support to retrieve the value based on the types?

Reply:
@wei-lee Do you mean like 4 functions like so?

```
string attr1 = user.getStringCustomAttribute(attributeName)
boolean attr2 = user.getBooleanCustomAttribute(attributeName)
int attr3 = user.getIntCustomAttribute(attributeName)
long attr4 = user.getLongCustomAttribute(attributeName)
```
I was thinking this might be a bit verbose initially, but if you think better typing would be better, I can update the PR.

Reply:
yes, I think we should have those methods. They should all call to the same method to retrieve the value, just need to parse the value based on the expected types.